src/components/Login.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+  <div class="login-container">
+    <div v-if="!isAuthenticated" class="login-section">
+      <div class="login-header">
+        <img src="../assets/icon.svg" alt="Dripplet Logo" class="logo" />
+        <h2>Welcome to Dripplet</h2>
+        <p>Sign in to start earning rewards</p>
+      </div>
+      <button @click="handleLogin" class="login-button" :disabled="loading">
+        <img src="../assets/google.svg" alt="Google" class="google-icon" v-if="!loading" />
+        <span v-if="loading" class="loading-spinner"></span>
+        {{ loading ? 'Signing in...' : 'Continue with Google' }}
+      </button>
+      <p v-if="error" class="error-message">{{ error }}</p>
+    </div>
+    <div v-else class="logged-in-section">
+      <div class="user-info" v-if="userProfile">
+        <img :src="userProfile.picture" alt="Profile" class="profile-picture" />
+        <div class="user-details">
+          <h3>{{ userProfile.name }}</h3>
+          <p>{{ userProfile.email }}</p>
+        </div>
+      </div>
+      <button @click="handleLogout" class="logout-button">
+        Sign Out
+      </button>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+
+interface UserProfile {
+  name: string
+  email: string
+  picture: string
+}
+
+const isAuthenticated = ref(false)
+const loading = ref(false)
+const error = ref<string | null>(null)
+const userProfile = ref<UserProfile | null>(null)
+
+const handleLogin = async () => {
+  loading.value = true
+  error.value = null
+
+  try {
+    const response = await new Promise((resolve) => {
+      chrome.runtime.sendMessage({ action: 'login' }, resolve)
+    })
+
+    if (response && response.success) {
+      isAuthenticated.value = true
+      await fetchUserProfile(response.accessToken)
+    } else {
+      error.value = response.error || 'Login failed. Please try again.'
+    }
+  } catch (err) {
+    error.value = 'An unexpected error occurred. Please try again.'
+  } finally {
+    loading.value = false
+  }
+}
+
+const handleLogout = async () => {
+  chrome.storage.local.remove(['accessToken', 'userProfile'], () => {
+    isAuthenticated.value = false
+    userProfile.value = null
+  })
+}
+
+const fetchUserProfile = async (accessToken: string) => {
+  try {
+    const response = await fetch('https://www.googleapis.com/oauth2/v2/userinfo', {
+      headers: {
+        Authorization: `Bearer ${accessToken}`
+      }
+    })
+    
+    if (!response.ok) {
+      throw new Error('Failed to fetch user profile')
+    }
+
+    const data = await response.json()
+    userProfile.value = {
+      name: data.name,
+      email: data.email,
+      picture: data.picture
+    }
+
+    // Store user profile
+    chrome.storage.local.set({ userProfile: userProfile.value })
+  } catch (err) {
+    console.error('Error fetching user profile:', err)
```

Comment:
We should also show some error message on the frontend here.

Reply:
displayed in ui